pipeline_steps/common.groovy

```diff
@@ -1704,40 +1704,136 @@ void createComponentSkeleton(String name, String repoUrl, String jiraProjectKey)
+    // been created in ${WORKSPACE}/rpc-gating.
+    git branch: 'master', url: 'https://github.com/rcbops/rpc-gating'
+
+    createComponentGateTrigger(name, repoUrl, jiraProjectKey)
+    createComponentPreRelease(name, repoUrl, releases, jiraProjectKey)
+
+    withEnv(
       [
-        string(
-          credentialsId: 'rpc-jenkins-svc-github-pat',
-          variable: 'PAT'
-        ),
-        usernamePassword(
-          credentialsId: "jira_user_pass",
-          usernameVariable: "JIRA_USER",
-          passwordVariable: "JIRA_PASS"
-        ),
+        "ISSUE_SUMMARY=Add component skeleton to ${name}",
+        "ISSUE_DESCRIPTION=This issue was generated automatically as part of registering a new component.",
+        "LABELS=component-skeleton jenkins",
+        "JIRA_PROJECT_KEY=${jiraProjectKey}",
+        "TARGET_BRANCH=master",
+        "COMMIT_TITLE=Add default jobs to ${name}",
+        "COMMIT_MESSAGE=This project is being added to the RE platform. This change adds some\ndefault jobs required by all projects.",
       ]
     ){
-      sshagent (credentials:['rpc-jenkins-svc-github-ssh-key']){
-        sh """#!/bin/bash -xe
-          set +x; . ${WORKSPACE}/.venv/bin/activate; set -x
-          ${WORKSPACE}/rpc-gating/scripts/add_component_gate_trigger_job.sh
-        """
-      }
-    }
-  }
+      withCredentials(
+        [
+          string(
+            credentialsId: 'rpc-jenkins-svc-github-pat',
+            variable: 'PAT'
+          ),
+          usernamePassword(
+            credentialsId: "jira_user_pass",
+            usernameVariable: "JIRA_USER",
+            passwordVariable: "JIRA_PASS"
+          ),
+        ]
+      ){
+        sshagent (credentials:['rpc-jenkins-svc-github-ssh-key']){
+          sh """#!/bin/bash -xe
+            set +x; . ${WORKSPACE}/.venv/bin/activate; set -x
+            git status
+            git diff
+            #${WORKSPACE}/rpc-gating/scripts/commit_and_pull_request.sh
+          """
+        } // sshagent
+      } // withCredentials
+    } // withEnv
+  } // dir
+}
+
+void createComponentGateTrigger(String name, String repoUrl, String jiraProjectKey){
+  String repo_dir = "${WORKSPACE}/rpc-gating-master"
+  String filename = "${name}.yml".replace("-", "_")
+  String projectsFile = "${repo_dir}/rpc_jobs/${filename}"
+  String jjb = """
+- project:
+    name: "${name}-gate-trigger"
+    repo_name: "${name}"
+    repo_url: "${repoUrl}"
+
+    jobs:
+      - 'Component-Gate-Trigger_{repo_name}'
+"""
+
+  dir(repo_dir) {
+    Boolean jobNotExists = sh(
+      returnStatus: true,
+      script: """#!/bin/bash -xe
+        grep -s 'Component-Gate-Trigger_{repo_name}' ${projectsFile}
+      """
+    ).asBoolean()
+
+    if (jobNotExists) {
+      sh """#!/bin/bash -xe
+        echo "${jjb}" >> ${projectsFile}
+      """
+    } // if
+  } // dir
+}
+
+void createComponentPreRelease(String name, String repoUrl, List releases, String jiraProjectKey){
```

Comment:
`jiraProjectKey` doesn't appear to be used by this function.

Reply:
Fixing